game/scripts/vscripts/items/sonic.lua

```diff
@@ -0,0 +1,48 @@
+item_sonic = class(TransformationBaseClass)
+
+LinkLuaModifier("modifier_item_sonic", "items/sonic.lua", LUA_MODIFIER_MOTION_NONE)
+LinkLuaModifier("modifier_sonic_fly", "modifiers/modifier_sonic_fly.lua", LUA_MODIFIER_MOTION_NONE)
+
+function item_sonic:GetIntrinsicModifierName()
+  return "modifier_item_sonic"
+end
+
+function item_sonic:GetTransformtionModifierName()
+  return "modifier_sonic_fly"
+end
+
+modifier_item_sonic = class(ModifierBaseClass)
+
+function modifier_item_sonic:IsHidden()
+  return true
+end
+
+function modifier_item_sonic:IsPurgable()
+  return false
+end
```

Comment:
Should also have `RemoveOnDeath` function returning false since it's an intrinsic modifier.